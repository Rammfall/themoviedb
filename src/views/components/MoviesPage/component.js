import React from 'react'
import PropTypes from 'prop-types'

import SearchForm from 'Views/components/SearchForm'
import PrivateLayout from 'Views/layouts/PrivateLayout'
import MoviesLoader from 'Views/components/MoviesLoader'
import MoviesList from 'Views/components/MoviesList'
import MoviesEmptyState from 'Views/components/MoviesEmptyState'
import RenderCtrl from 'Views/components/RenderCtrl'

const MoviesPage = ({
  movies,
  isLoading,
  isEmpty
}) => (
  <PrivateLayout>
    <SearchForm />
    <div className='top-margin'>
      <RenderCtrl
        isLoading={isLoading}
        isEmpty={isEmpty}
        renderEmpty={<MoviesEmptyState />}
        renderLoading={<MoviesLoader />}
        renderList={(
          <MoviesList
            movies={movies}
          />
        )}
      />
    </div>
  </PrivateLayout>
)

MoviesPage.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({
    original_title: PropTypes.string,
    id: PropTypes.number,
    overview: PropTypes.string,
    poster_path: PropTypes.string
  })).isRequired,
  isLoading: PropTypes.bool,
  isEmpty: PropTypes.bool.isRequired
}

MoviesPage.defaultProps = {
  isLoading: null
}

export default MoviesPage
