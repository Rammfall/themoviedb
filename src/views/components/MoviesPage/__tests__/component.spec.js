import React from 'react'
import { shallow } from 'enzyme'

import MoviesPage from '../component'

describe('MoviesPage', () => {
  it('matches snapshot', () => {
    const props = {
      isLoading: false,
      isEmpty: false
    }
    const component = shallow(<MoviesPage {...props} />)

    expect(component).toMatchSnapshot()
  })
})
