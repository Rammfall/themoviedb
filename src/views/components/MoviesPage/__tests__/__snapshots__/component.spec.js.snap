// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesPage matches snapshot 1`] = `
<PrivateLayout>
  <withRouter(Connect(WithFormik(SearchForm))) />
  <div
    className="top-margin"
  >
    <RenderCtrl
      isEmpty={false}
      isLoading={false}
      renderEmpty={<MoviesEmptyState />}
      renderList={
        <MoviesList
          movies={
            Array [
              Object {
                "id": 1,
                "original_title": "test",
                "overview": "test",
                "poster_path": "uri",
              },
            ]
          }
        />
      }
      renderLoading={<MoviesLoader />}
    />
  </div>
</PrivateLayout>
`;
